File: AFront/src/app/Entity/Notificacion.ts
export interface Notificacion {
  id: number;
  clienteId: number;
  mensaje: string;
  fechaEnvio: Date;
  tipo: TipoNotificacion;
  leido: boolean;
  fechaCreacion: Date;
  fechaActualizacion: Date | null;
  fechaBorrado: Date | null;
  estaBorrado: boolean;
}


export enum TipoNotificacion {
  General = 0,
  Factura = 1,
  Carrito = 2,
  Servicio = 3,
  NuevaFactura = 4,
  RecordatorioPago = 5,
}

File: models/Notificacion.cs
namespace TallerMecanico.models;

public class Notificacion
{
    public int Id { get; set; }
    public int ClienteId { get; set; }
    public Cliente Cliente { get; set; }
    public string Mensaje { get; set; }
    public DateTime FechaEnvio { get; set; }
    public TipoNotificacion Tipo { get; set; }
    public bool Leido { get; set; }

    // Auditoría
    public DateTime FechaCreacion { get; set; }
    public DateTime? FechaActualizacion { get; set; }
    public DateTime? FechaBorrado { get; set; }
    public bool EstaBorrado { get; set; } = false;
}

public enum TipoNotificacion
{
    General,
    Factura,
    Carrito,
    Servicio,
    NuevaFactura,
    RecordatorioPago
}
